homeassistant/components/google_photos/media_source.py

```diff
@@ -25,27 +25,48 @@
 # photos when displaying the users library. We fetch a minimum of 50 photos
 # unless we run out, but in pages of 100 at a time given sometimes responses
 # may only contain a handful of items Fetches at least 50 photos.
-MAX_PHOTOS = 50
+MAX_RECENT_PHOTOS = 50
+MAX_ALBUMS = 50
 PAGE_SIZE = 100
 
 THUMBNAIL_SIZE = 256
 LARGE_IMAGE_SIZE = 2160
 
 
-# Markers for parts of PhotosIdentifier url pattern.
+@dataclass
+class SpecialAlbumDetails:
+    """Details for a Special album."""
+
+    path: str
+    title: str
+    list_args: dict[str, Any]
+    max_photos: int | None
+
+
+class SpecialAlbum(Enum):
+    """Special Album types."""
+
+    RECENT = SpecialAlbumDetails("recent", "Recent Photos", {}, MAX_RECENT_PHOTOS)
+    FAVORITE = SpecialAlbumDetails(
+        "favorites", "Favorite Photos", {"favorites": True}, None
```

Comment:
Why no max_photos for favorite? Is it because whenever media source supports pagination, recent will have a single page but favorites will have multiple?
Also for recent below you potentially issue multiple requests if the server returned less than 50, so (assuming you have more than 50 in your library) you will get from 50 to 149 items. But for favorites (assuming you have more than 100 favorites in your library) you will get from 0 (e.g. everything in the page filtered out on server side) to 100 items.

Reply:
I was assuming that # of favorites would be much smaller than # of items and it would be reasonable. This inconsistency didn't really bother me since in the worst case you get a little more photos even though it's not all exact. It's messy/hacky, but I thought it might be ok?